🐼 Pandas Tutorial › Pandas Tutorial &nbsp;·&nbsp; *Bài 8/15*

[« Pandas Read JSON](007_pandas_read_json.ipynb) &nbsp;•&nbsp; [📚 Mục lục](../../README.md) &nbsp;•&nbsp; [Pandas - Cleaning Data »](009_pandas_cleaning_data.ipynb)

# Pandas - Analyzing DataFrames

> 📖 **Học song song trên W3Schools:** [pandas_analyzing.asp](https://www.w3schools.com/python/pandas/pandas_analyzing.asp)
>
> 🎯 **Trong bài này:** Viewing the Data · Info About the Data · Result Explained · Null Values · ➕ Mở rộng: describe() và isnull()

## Viewing the Data — Xem nhanh dữ liệu

Một trong những phương thức hay dùng nhất để xem nhanh DataFrame là `head()`.

`head()` trả về **tiêu đề** và **số hàng chỉ định**, tính từ đầu bảng.

Lấy 10 hàng đầu:

In [1]:
import pandas as pd

df = pd.read_csv('data.csv')

print(df.head(10))

   Duration  Pulse  Maxpulse  Calories
0        60     92       137     394.8
1        90     98       116     483.6
2        20    109       123     119.4
3        60    107       144     346.3
4       150     94       119     719.8
5        60    149       186     232.0
6        60    105       126     326.3
7       120    114       135     479.7
8       180     96       128     819.8
9        60     95       105     397.1


> 📌 Nếu không chỉ định số hàng, `head()` trả về **5** hàng đầu.

In [2]:
import pandas as pd

df = pd.read_csv('data.csv')

print(df.head())

   Duration  Pulse  Maxpulse  Calories
0        60     92       137     394.8
1        90     98       116     483.6
2        20    109       123     119.4
3        60    107       144     346.3
4       150     94       119     719.8


Phương thức `tail()` trả về các hàng **cuối cùng**:

In [3]:
import pandas as pd

df = pd.read_csv('data.csv')

print(df.tail())

     Duration  Pulse  Maxpulse  Calories
164        60     82       111     376.3
165        60     93       118     368.3
166        60     80       101     291.4
167        60    133       154     320.6
168        60    121       139     324.4


## Info About the Data — Thông tin về dữ liệu

Phương thức `info()` cung cấp thêm thông tin về tập dữ liệu:

In [4]:
import pandas as pd

df = pd.read_csv('data.csv')

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 169 entries, 0 to 168
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Duration  169 non-null    int64  
 1   Pulse     169 non-null    int64  
 2   Maxpulse  169 non-null    int64  
 3   Calories  164 non-null    float64
dtypes: float64(1), int64(3)
memory usage: 5.4 KB


## Result Explained — Giải thích kết quả

Kết quả cho biết:

- Có **169 hàng** và **4 cột**:

```text
RangeIndex: 169 entries, 0 to 168
Data columns (total 4 columns):
```

- Tên mỗi cột và kiểu dữ liệu:

```text
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   Duration  169 non-null    int64
 1   Pulse     169 non-null    int64
 2   Maxpulse  169 non-null    int64
 3   Calories  164 non-null    float64
```

## Null Values — Giá trị rỗng

`info()` còn cho biết **mỗi cột có bao nhiêu giá trị không rỗng (Non-Null)**.

Cột "Calories" chỉ có **164 / 169** giá trị — tức là có **5 hàng** không có giá trị Calories.

Giá trị rỗng (Null) có thể làm sai kết quả phân tích. Nên cân nhắc **xoá** các hàng đó — bước này gọi là **làm sạch dữ liệu**, bạn sẽ học ở các bài sau.

## ➕ Mở rộng: describe() và isnull()

In [5]:
import pandas as pd

df = pd.read_csv('data.csv')
print(df.describe().round(1))      # thống kê nhanh: count, mean, std, min, max...
print(df.isnull().sum())           # số giá trị rỗng mỗi cột

       Duration  Pulse  Maxpulse  Calories
count     169.0  169.0     169.0     164.0
mean       84.3  106.9     130.1     463.6
std        65.1   15.6      18.3     363.0
min        20.0   80.0      87.0     103.1
25%        45.0   95.0     118.0     259.6
50%        60.0  108.0     128.0     330.1
75%        90.0  118.0     142.0     482.2
max       300.0  149.0     186.0    1847.4
Duration    0
Pulse       0
Maxpulse    0
Calories    5
dtype: int64


---

## 🧠 Ghi nhớ nhanh

> - `df.head(n)` n hàng đầu (mặc định 5) · `df.tail(n)` n hàng cuối.
> - `df.info()` → số hàng, tên cột, kiểu dữ liệu, số giá trị **Non-Null**.
> - `df.describe()` thống kê nhanh · `df.isnull().sum()` đếm giá trị rỗng.

---

## 📝 Exercise — Bài tập nhanh

**❓ Câu hỏi:** `df.head()` không có tham số trả về bao nhiêu hàng?

- **A.** 10
- **B.** 5
- **C.** 3
- **D.** 1

<details>
<summary>👉 Xem đáp án</summary>

✅ **Đáp án: B** — 5

</details>

---

[« Pandas Read JSON](007_pandas_read_json.ipynb) &nbsp;•&nbsp; [📚 Mục lục](../../README.md) &nbsp;•&nbsp; [Pandas - Cleaning Data »](009_pandas_cleaning_data.ipynb)